service/config_db.hpp

```diff
@@ -0,0 +1,262 @@
+//
+// Created by riz on 29/01/19.
+//
+
+#pragma once
+
+#include <cstring>
+#include <vector>
+#include <sstream>
+#include <sqlite_modern_cpp.h>
+
+namespace raven::db
+{
+  template<typename T>
+  class blob
+  {
+  public:
+    //! getters
+    template<typename vtype>
+    std::enable_if_t<std::is_fundamental<vtype>::value, vtype> get_value() const noexcept
+    {
+        vtype ret;
+        std::memcpy(&ret, data_.data(), sizeof(vtype));
+        return ret;
+    }
+
+    // le compilo me laisse pas faire une template specifier pour std::string je sais pas trop pourquoi ( je suis surement un abrutit mais la il est tard je suis fatigué) à investiguer
+    std::string get_value_string() const noexcept
+    {
+        return (std::string(data_.begin(), data_.end()));
+    }
+
+    const std::vector<T> &get_data() const noexcept
+    {
+        return data_;
+    }
+
+    std::vector<T> &get_data() noexcept
+    {
+        return data_;
+    }
+
+    //! Constructors
+    template<typename vtype, typename std::enable_if_t<std::is_fundamental<T>::value, int> = 0>
+    explicit blob(vtype value) noexcept
+    {
+        data_.resize(sizeof(vtype) / sizeof(T));
+        std::memcpy(data_.data(), &value, sizeof(vtype));
+    }
+
+    explicit blob(std::vector<T> value) noexcept
+    :data_{std::move(value)}
+    {
+    }
+
+    explicit blob(const std::string &value) noexcept
+    :data_{value.begin(), value.end()}
+    {
+    }
+
+    explicit blob(const char *value) noexcept
+    :blob{std::string(value)}
+    {
+    }
+
+  private:
+    std::vector<T> data_;
+  };
+
+  template<typename T>
+  inline sqlite::database_binder &operator<<(sqlite::database_binder &db, const blob<T> &value)
+  {
+    return db << value.get_data();
+  }
+
+  class config_db
+  {
+  public:
+    //! Operations
+    long create_config(const std::string &name)
+    {
+        int maxId;
+        database_ << "select coalesce(max(id), 0) from configIndex;" >> maxId; // TODO: change to get real increment count
+        std::ostringstream new_table_name("");
+        new_table_name << name << "_" << maxId + 1;
+        database_ << "create table if not exists " + new_table_name.str() + " (" +
```

Comment:
-- auto-generated definition
create table if not exists config
(
  CONFIG_NAME TEXT,
  id          INTEGER not null
    constraint config_pk
      primary key autoincrement
);

create unique index config_id_uindex
  on config (id);

This is the good way to do this.

Reply:
What does `constraint config_pk` ?